apps/04TowersOfHanoi.js

```diff
@@ -13,32 +13,103 @@ var stacks = {
   c: []
 };
 
+var moves = 0;
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
   // Your code here
 
+  // popping an item from the startStack into the endStack
+  var block = stacks[startStack].pop();
+  console.log('popped block', block);
+  stacks[endStack].push(block);
+
+  moves++;
+  if (moves < 2) {
+    console.log('you made ' + moves + " move")
+  } else {
+    console.log('you made ' + moves + " moves");
+  }
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
   // Your code here
+  //checking to see if you can move
+  if (stacks[startStack].length === 0) {
+    return false;
+  }
+  //moving the piece if the move is legal
+  if (stacks[endStack].length === 0) {
+    return true;
+  }
+
+  var startBlock = stacks[startStack][stacks[startStack].length - 1]
+  var endBlock = stacks[endStack][stacks[endStack].length - 1]
+  if (startBlock < endBlock) {
+    return true;
+  } else {
+    return false;
+    console.log('illegal move!');
+  }
+}
 
+//a,b, or c
+//numbers
+//capital letters
+//strings vs variables
+function isInputValid(startStack, endStack) {
+  //convert to lowercase and check for right input
+  let start = startStack.toLowerCase();
+  let end = endStack.toLowerCase();
+  if (start === 'a' || start === 'b' || start === 'c')
+    if (end === 'a' || end === 'b' || end === 'c') {
+      return true;
+    }
+  return false;
 }
 
-function checkForWin() {
-  // Your code here
 
-}
 
 function towersOfHanoi(startStack, endStack) {
   // Your code here
+  if (isInputValid(startStack, endStack)) {
+    getPrompt();
```

Comment:
Getting a little tricky here. Keep an eye on your return values for this function, we want them to be consistent in all cases